packages/web/src/components/ui/SidebarCategory/SidebarCategorySearch.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { ScanResultPackageWithMetadata } from '@gradejs-public/public-api/src/clientApiRouter';
+import SidebarCategoryHeader from './SidebarCategoryHeader';
+import Search from '../Search/Search';
+import { SearchText } from 'store/slices/scanDisplayOptions';
+
+type Props = {
+  categoryName: string;
+  packages: ScanResultPackageWithMetadata[] | undefined;
+  searchText: SearchText;
+  handleSearchTextChange: (newSearchText: SearchText) => void;
+};
+
+export default function SidebarCategorySearch({
+  categoryName,
+  searchText,
+  handleSearchTextChange,
+}: Props) {
+  const [searchValue, setSearchValue] = useState<string>('');
```

Comment:
Why do need the internal state here? The whole component looks like a simple wrapper.

Reply:
To show user input while typing as-is and to update redux state via debounced callback after user is done typing to cut unnecessary and expensive sort/filter recalculations on every keystroke.

Reply:
Seems like something that should be handled higher in component tree, rather than at a specialized input level.